precompile/contract/contract.go

```diff
@@ -80,5 +101,10 @@ func (s *statefulPrecompileWithFunctionSelectors) Run(accessibleState Accessible
 		return nil, suppliedGas, fmt.Errorf("invalid function selector %#x", selector)
 	}
 
+	// Check if the function is activated
+	if !function.IsActivated(accessibleState) {
+		return nil, suppliedGas, fmt.Errorf("invalid non-activated function selector %#x", selector)
```

Comment:
Note: the error message does not change the rules of processing the block itself although the return value when an error is returned does.